### ISSUE TITLE
ERROR using SNAP 10

### ISSUE THREAD

**Issue by anikaweinmann** [actinia-org/actinia-docker#83]
When trying to update to `SNAP 10` (see https://github.com/mundialis/esa-snap/pull/41), the following error occurred:
```
i.sentinel1.pyrosargeocode -fndm input=/grassdb/....SAFE.zip output_directory=/grassdb/tmp_s1/out aoi=/grassdb/tmp_s1/aoi.geojson elevation=dem temporary_directory=/grassdb/tmp_s1/tmp/ polarizations=VV auxillary_data=layoverShadowMask nprocs=1 --v

Start geocoding scene
....SAFE.zip
ERROR: Geocoding failed with the following error: cannot read version
       information from /root/.snap/system/var/log/messages.log.
       Please restart SNAP.
       Please check the log files
```
In the log file the version is listed as:
```
  Product Version         = SNAP (Build 11.3-6b879cb782eaa4f13a731aff82eada11289a66f7)
  Operating System        = Linux version 5.15.0-83-generic running on amd64
  Java; VM; Vendor        = 11.0.23; OpenJDK 64-Bit Server VM 11.0.23+9-alpine-r0; Alpine
  Runtime                 = OpenJDK Runtime Environment 11.0.23+9-alpine-r0
  Java Home               = /usr/lib/jvm/java-11-openjdk
  System Locale; Encoding = en_gb (snap); UTF-8
  Home Directory          = /root
  Current Directory       = /
  User Directory          = /root/.snap/system
  Cache Directory         = /root/.snap/system/var/cache
  Installation            = /usr/local/snap/snap
                            /usr/local/snap/platform
                            /usr/local/snap/etc
                            /usr/local/snap/ide
                            /usr/local/snap/microwavetbx
                            /usr/local/snap/rstb
                            /usr/local/snap/smos
                            /usr/local/snap/platform
```
The installed `pyroSAR` version is too old (0.24.0), but alpine 3.18 does not install the new version (0.26.0). Only [`pyronSAR>=0.26.0`](https://github.com/johntruckenbrodt/pyroSAR/releases/tag/v0.26.0) ist compatibe with SNAP 10.

For the new SNAP version `openjdk11` has to be installed to change [here](https://github.com/actinia-org/actinia-docker/blob/main/actinia-alpine/Dockerfile_alpine_dependencies#L27).

**Comment by ninsbl** [comment]
Hmm. We have not moved to SNAP 10 yet, and we actually use actinia-ubuntu because our workflows require pytorch which we could not get working in alpine...
So I do not have any suggestions here, sorry... Will come back to you when we move to SNAP 10 on Ubuntu...

(BTW: Ubuntu builds are much faster and the size advantage of alpine is almost gone once heavy packages like pytorch and torchvision are installed...).

**Comment by ninsbl** [comment]
Is that still an issue?

I am about to move to SNAP 10 on UBUNTU as well.
However, I tried to install _pyrosar>=0.26_ in _Alpine 3.18_ and it seems to work fine in this test-docker-file:

```Dockerfile
FROM alpine:3.18 as version
RUN apk add python3 py3-pip postgresql-dev gcc python3-dev musl-dev
RUN python3 -m pip install --break-system-packages pyrosar>=0.26
```

**Comment by anikaweinmann** [comment]
Yes it is still an issue. We have not tried it again. We are using now `alpine:3.20` as base for actinia.
Thank you then we will try if it works also on Alpine 3.20.

**Comment by ninsbl** [comment]
```Dockerfile
FROM alpine:3.20 as version
RUN apk add python3 py3-pip postgresql-dev gcc python3-dev musl-dev
RUN python3 -m pip install --break-system-packages pyrosar>=0.26
```
seems to work fine as well...

**Comment by mmacata** [comment]
Indeed the pyrosar version with above commands would still have pyrosar 0.24.0 installed without notice or warning!
Reason most likely is the required python version defined [here](https://github.com/johntruckenbrodt/pyroSAR/blob/main/pyproject.toml#L7) `requires-python = ">=3.8,<3.11"` and alpine having python 3.12 already.
The following command does the trick:

`pip install --ignore-requires-python pyroSAR`

**Comment by johntruckenbrodt** [comment]
Hi. I am the maintainer of pyroSAR. Unfortunately I had to place an upper limit on the Python version in https://github.com/johntruckenbrodt/pyroSAR/pull/293. From version 0.25.0 on, pyroSAR does have this limit. I did not however fix all previous versions.  I understand this is inconvenient. If python>=3.11 is used, an older version of pyroSAR is installed and users don't see what's happening.  
A solution for the problem with `sqlalchemy_utils` seems to have been found. I just have not found the time to check whether it's working for pyroSAR. I try to do this soon.  
When ignoring the python constraint during installation, the database functionality will not work. If this is not needed, this might be a possible workaround.

**Comment by neteler** [comment]
Thanks for you feedback, @johntruckenbrodt! 
Seems that this Python limitation only affects Windows users (at least, what I understand from https://github.com/kvesteri/sqlalchemy-utils/issues/716#issuecomment-2360973556)?

**Comment by johntruckenbrodt** [comment]
The newly released version 0.27.0 should fix it.
